pkg/linter/lintFilterRuleSplitSubnetCommon.go

```diff
@@ -14,56 +14,41 @@ import (
 	"github.com/np-guard/vpc-network-config-analyzer/pkg/vpcmodel"
 )
 
-const splitRuleSubnetName = "rules-splitting-subnets"
-
-// filterRuleSplitSubnetLint: rules of filters that are inconsistent w.r.t. subnets.
-type filterRuleSplitSubnetLint struct {
-	basicLinter
-}
-
 // a rule with the list of subnets it splits
 type splitRuleSubnet struct {
 	rule         vpcmodel.RuleOfFilter
 	splitSubnets []vpcmodel.Subnet
 }
 
-// /////////////////////////////////////////////////////////
-// lint interface implementation for filterRuleSplitSubnetLint
-// ////////////////////////////////////////////////////////
-func (lint *filterRuleSplitSubnetLint) lintName() string {
-	return splitRuleSubnetName
-}
-
-func (lint *filterRuleSplitSubnetLint) lintDescription() string {
-	return "Firewall rules implying different connectivity for different endpoints within a subnet"
-}
+////////////////////////////////////////////////////////////////////////////////////////////
+// functionality used by both filterRuleSplitSubnetLintNACL and filterRuleSplitSubnetLintSG
+////////////////////////////////////////////////////////////////////////////////////////////
 
-func (lint *filterRuleSplitSubnetLint) check() error {
-	for _, config := range lint.configs {
+func findSplitRulesSubnet(configs map[string]*vpcmodel.VPCConfig, filterLayerName string) (res []splitRuleSubnet, err error) {
+	res = []splitRuleSubnet{}
```

Comment:
```suggestion
```
nit: this line is redundant. `res` is already initialized (to `nil`, but that's ok)